src/components/application_manager/include/application_manager/application_manager_impl.h

```diff
@@ -281,24 +202,14 @@ class ApplicationManagerImpl
    */
   void ChangeAppsHMILevel(uint32_t app_id, mobile_apis::HMILevel::eType level);
 
-  void Erase(ApplicationSharedPtr app_to_remove) {
-    DCHECK(app_to_remove);
-    app_to_remove->RemoveExtensions();
-    applications_.erase(app_to_remove);
-  }
-
-  virtual functional_modules::PluginManager& GetPluginManager() OVERRIDE {
-    return plugin_manager_;
+  virtual plugin_manager::RPCPluginManager& GetPluginManager() OVERRIDE {
+    DCHECK(plugin_manager_);
+    return *plugin_manager_;
```

Comment:
shall the following be in vehicle info plugin? 


 /**
   * @brief Notifies all components interested in Vehicle Data update
   * i.e. new value of odometer etc and returns list of applications
   * subscribed for event.
   * @param vehicle_info Enum value of type of vehicle data
   * @param new value (for integer values currently) of vehicle data
   */
  std::vector<ApplicationSharedPtr> IviInfoUpdated(
      mobile_apis::VehicleDataType::eType vehicle_info, int value) OVERRIDE;

Reply:
@yang1070 please look at a4fb9b9896d548e927a070d80a3a5e91c1feef7b
we updated this function and now it just notifies policy about updating